In [6]:
# import all the filenames
from allfunctions import *

In [8]:
%load_ext autoreload
%autoreload 2

The autoreload extension is already loaded. To reload it, use:
  %reload_ext autoreload


1  def LassoCVModel(filename):

2  def OMPCVModel(filename):

3  def GradientBoostingCVModel(filename):

4  def RandomForestCVModel(filename):

5  def RidgeCVModel(filename):

6  def ElasticNetCVModel(filename):

7  def SVRPolyCVModel(filename):

8  def SVRSigmoidCVModel(filename):

9  def SVRLinearCVModel(filename):

10 def SVRRbfCVModel(filename):

In [2]:
functionList = [OMPCVModel]# [ LassoCVModel, OMPCVModel, GradientBoostingCVModel ] 
''', , 
                 RandomForestCVModel, RidgeCVModel, ElasticNetCVModel, 
                 SVRPolyCVModel, SVRSigmoidCVModel, SVRLinearCVModel, 
                 SVRRbfCVModel ] #OMPCVModel
'''

', , \n                 RandomForestCVModel, RidgeCVModel, ElasticNetCVModel, \n                 SVRPolyCVModel, SVRSigmoidCVModel, SVRLinearCVModel, \n                 SVRRbfCVModel ] #OMPCVModel\n'

In [7]:
def wrapper(functionList = [], filenames = [ 'datasetIPPD.pickle' ]):
    if __name__ == '__main__':
        global model_selection
                
        model_selection = {}
        
        for f in functionList:
            print()
            for filename in filenames:
                out = f(filename)
                newScore = out[filename]['train_rmse_cv_4_picking']

                if f in model_selection:
                    #compare scores and select more accurate model
                    oldScore = model_selection[f]['train_rmse_cv_4_picking']
                    if(newScore < oldScore):
                        print("【【【】】】switching models for %s..." % filename)
                        print("【【【】】】newScore: %f < oldScore: %f" % (newScore, oldScore))
                        print("【【【】】】newModel: %s \n" % f)
                        print(100*'_')
                        model_selection[f] = out[filename]
                else:
                    #no score to compare
                    print("【【】】adding new model for %s \n" % filename)
                    print("【【【】】】newScore: %f " % (newScore))
                    print(100*'_')
                    model_selection[f] = out[filename]
    return model_selection

<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:><:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>
<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:><:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>
<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:><:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>
<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:><:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>


# 1 LassoCVModel--OK

In [83]:
t_lasso = wrapper([LassoCVModel])


Dataset size read: train 256 and test 65 

# Tuning hyper-parameters 

Starting grdsurch.fit(X_train, y_train)

Best parameters set found on development set:

{'alpha': 20.433597178569439, 'precompute': True}
Lasso(alpha=20.433597178569439, copy_X=True, fit_intercept=True,
   max_iter=10000000.0, normalize=False, positive=False, precompute=True,
   random_state=None, selection='random', tol=1e-06, warm_start=False)

【【】】adding new model for datasetIPPD.pickle 

【【【】】】newScore: 1025.086098 
____________________________________________________________________________________________________


In [131]:
for key in t_lasso.keys():
    print(key)
    print(t_lasso[key]['model'],'\n\n\n\n',
          t_lasso[key]['train_rmse_cv_4_picking'],
          t_lasso[key]['test_rmse_4_reporting'],
          t_lasso[key]['test_mean_y_4_comparing']
         )
    print(100*'_')

<function LassoCVModel at 0x7f4f4c57dc80>
Lasso(alpha=20.433597178569439, copy_X=True, fit_intercept=True,
   max_iter=10000000.0, normalize=False, positive=False, precompute=True,
   random_state=None, selection='random', tol=1e-06, warm_start=False) 



 1025.08609776 880.826424842 1532.34615385
____________________________________________________________________________________________________


In [132]:
for k in t_lasso.keys():
    print(k)
    mo = t_lasso[k]['model']

<function LassoCVModel at 0x7f4f4c57dc80>


In [141]:
names = ['ppn', 'sizeGB', 'ft1', 'ft2', 'ft3', 'ft4', 'ft5', 'ft6', 'ft7', 'ft8',
        'ft9', 'ft10', 'ft11', 'ft12', 'ft13', 'ft14', 'ft15', 'ft16', 'ft17',
        'ft18', 'ft24', 'ft25', 'ft19', 'ft39', 'ft20', 'ft33', 'ft21', 'ft35',
        'ft22', 'ft23', 'ft34', 'ft36', 'ft37', 'ft38', 'ft40', 'ft41', 'ft42',
        'ft43', 'ft44', 'ft45', 'ft46', 'ft48', 'ft47', 'ft49', 'ft51', 'ft50',
        'ft52', 'ft53', 'ft54', 'ft55', 'ft26', 'ft27', 'ft28', 'ft29', 'ft30',
        'ft31', 'ft32']

In [145]:
print(sorted(zip(map(lambda x: x, mo.coef_), 
                 names), reverse=True))

[(-0.0, 'sizeGB'), (-0.0, 'ppn'), (-0.0, 'ft9'), (-0.0, 'ft8'), (-0.0, 'ft7'), (0.0, 'ft6'), (-0.0, 'ft55'), (0.0, 'ft54'), (0.0, 'ft53'), (0.0, 'ft52'), (0.0, 'ft51'), (0.0, 'ft50'), (0.0, 'ft5'), (-0.0, 'ft49'), (-0.0, 'ft48'), (0.0, 'ft47'), (0.0, 'ft46'), (-0.0, 'ft45'), (0.0, 'ft44'), (0.0, 'ft43'), (-0.0, 'ft42'), (0.0, 'ft41'), (-0.0, 'ft40'), (0.0, 'ft4'), (0.0, 'ft39'), (0.0, 'ft38'), (0.0, 'ft37'), (-0.0, 'ft36'), (0.0, 'ft35'), (0.0, 'ft34'), (-0.0, 'ft33'), (0.0, 'ft32'), (-0.0, 'ft31'), (-0.0, 'ft30'), (-0.0, 'ft3'), (0.0, 'ft29'), (-0.0, 'ft28'), (0.0, 'ft27'), (0.0, 'ft26'), (-0.0, 'ft25'), (0.0, 'ft24'), (0.0, 'ft23'), (0.0, 'ft22'), (0.0, 'ft21'), (0.0, 'ft20'), (-0.0, 'ft2'), (0.0, 'ft19'), (0.0, 'ft18'), (-0.0, 'ft17'), (-0.0, 'ft16'), (0.0, 'ft15'), (0.0, 'ft14'), (0.0, 'ft13'), (0.0, 'ft12'), (-0.0, 'ft11'), (0.0, 'ft10'), (-0.0, 'ft1')]


#>>>>>>>>>>>


# Lasso Final

Lasso(alpha=20.433597178569439, copy_X=True, fit_intercept=True,
   max_iter=10000000.0, normalize=False, positive=False, precompute=True,
   random_state=None, selection='random', tol=1e-06, warm_start=False) 



 1025.08609776 880.826424842 1532.34615385

<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:><:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>
<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:><:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>
<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:><:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>
<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:><:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>


# 2 OMPCVModel --OK

In [87]:
t_OMPC = wrapper([OMPCVModel])


Dataset size read: train 256 and test 65 

# Tuning hyper-parameters 

Starting grdsurch.fit(X_train, y_train)


/data/home/a1singh/anaconda3/lib/python3.6/site-packages/sklearn/linear_model/omp.py:665: RuntimeWarning:  Orthogonal matching pursuit ended prematurely due to linear
dependence in the dictionary. The requested precision might not have been met.

  return_n_iter=True)
/data/home/a1singh/anaconda3/lib/python3.6/site-packages/sklearn/linear_model/omp.py:665: RuntimeWarning:  Orthogonal matching pursuit ended prematurely due to linear
dependence in the dictionary. The requested precision might not have been met.

  return_n_iter=True)
/data/home/a1singh/anaconda3/lib/python3.6/site-packages/sklearn/linear_model/omp.py:665: RuntimeWarning:  Orthogonal matching pursuit ended prematurely due to linear
dependence in the dictionary. The requested precision might not have been met.

  return_n_iter=True)
/data/home/a1singh/anaconda3/lib/python3.6/site-packages/sklearn/linear_model/omp.py:665: RuntimeWarning:  Orthogonal matching pursuit ended prematurely due to linear
dependence in the dictiona


Best parameters set found on development set:

{'n_nonzero_coefs': 3, 'tol': 1e-20}
OrthogonalMatchingPursuit(fit_intercept=True, n_nonzero_coefs=3,
             normalize=True, precompute='auto', tol=1e-20)

【【】】adding new model for datasetIPPD.pickle 

【【【】】】newScore: 445.063184 
____________________________________________________________________________________________________


/data/home/a1singh/anaconda3/lib/python3.6/site-packages/sklearn/linear_model/omp.py:665: RuntimeWarning:  Orthogonal matching pursuit ended prematurely due to linear
dependence in the dictionary. The requested precision might not have been met.

  return_n_iter=True)


In [146]:
for key in t_OMPC.keys():
    print(key)
    omp_mo = t_OMPC[key]['model']
    print(t_OMPC[key]['model'],'\n\n\n',
          t_OMPC[key]['train_rmse_cv_4_picking'],
          t_OMPC[key]['test_rmse_4_reporting'],
          t_OMPC[key]['test_mean_y_4_comparing']
         )

    print(100*'_')

<function OMPCVModel at 0x7f4f23fe96a8>
OrthogonalMatchingPursuit(fit_intercept=True, n_nonzero_coefs=3,
             normalize=True, precompute='auto', tol=1e-20) 


 445.063183594 354.963614472 1532.34615385
____________________________________________________________________________________________________


#>>>>>>>>>>>


# OMP Final

OrthogonalMatchingPursuit(fit_intercept=True, n_nonzero_coefs=3,
             normalize=True, precompute='auto', tol=1e-20) 


 445.063183594 354.963614472 1532.34615385

In [150]:
print(omp_mo.n_nonzero_coefs_)
print()
print(sorted(zip(map(lambda x: x, omp_mo.coef_), 
                 names), reverse=True))

3

[(302.3673068598909, 'sizeGB'), (98.0619727553009, 'ft39'), (3.3961833980045406, 'ppn'), (0.49317894769754, 'ft51'), (0.0, 'ft9'), (0.0, 'ft8'), (0.0, 'ft7'), (0.0, 'ft6'), (0.0, 'ft55'), (0.0, 'ft54'), (0.0, 'ft53'), (0.0, 'ft52'), (0.0, 'ft50'), (0.0, 'ft5'), (0.0, 'ft49'), (0.0, 'ft48'), (0.0, 'ft47'), (0.0, 'ft46'), (0.0, 'ft45'), (0.0, 'ft44'), (0.0, 'ft43'), (0.0, 'ft42'), (0.0, 'ft41'), (0.0, 'ft4'), (0.0, 'ft38'), (0.0, 'ft37'), (0.0, 'ft36'), (0.0, 'ft35'), (0.0, 'ft34'), (0.0, 'ft33'), (0.0, 'ft32'), (0.0, 'ft31'), (0.0, 'ft30'), (0.0, 'ft3'), (0.0, 'ft29'), (0.0, 'ft28'), (0.0, 'ft27'), (0.0, 'ft26'), (0.0, 'ft25'), (0.0, 'ft24'), (0.0, 'ft23'), (0.0, 'ft22'), (0.0, 'ft21'), (0.0, 'ft20'), (0.0, 'ft2'), (0.0, 'ft19'), (0.0, 'ft18'), (0.0, 'ft17'), (0.0, 'ft16'), (0.0, 'ft15'), (0.0, 'ft14'), (0.0, 'ft13'), (0.0, 'ft12'), (0.0, 'ft11'), (0.0, 'ft10'), (0.0, 'ft1'), (-4.4203725761317223, 'ft40')]


#>>>>>>>>>>>


# Final Model : OMPC

function OMPCVModel at 0x7f4f23fe96a8>
OrthogonalMatchingPursuit(fit_intercept=True, n_nonzero_coefs=3,
             normalize=True, precompute='auto', tol=1e-20) 


 445.063183594 354.963614472 1532.34615385

# 3 GradientBoostingCVModel -- OK

In [70]:
t_boosting = wrapper([GradientBoostingCVModel])


Dataset size read: train 256 and test 65 

# Tuning hyper-parameters 

Starting grdsurch.fit(X_train, y_train)

Best parameters set found on development set:

{'learning_rate': 0.005, 'loss': 'ls', 'n_estimators': 6000}
GradientBoostingRegressor(alpha=0.9, criterion='friedman_mse', init=None,
             learning_rate=0.005, loss='ls', max_depth=None,
             max_features=None, max_leaf_nodes=None,
             min_impurity_split=1e-07, min_samples_leaf=1,
             min_samples_split=2, min_weight_fraction_leaf=0.0,
             n_estimators=6000, presort='auto', random_state=None,
             subsample=1.0, verbose=0, warm_start=False)

【【】】adding new model for datasetIPPD.pickle 

【【【】】】newScore: 421.208898 
____________________________________________________________________________________________________


In [71]:
for key in t_boosting.keys():
    print(key)
    print(
          t_boosting[key]['model'],'\n\n\n',
          t_boosting[key]['train_rmse_cv_4_picking'],
          t_boosting[key]['test_rmse_4_reporting'],
          t_boosting[key]['test_mean_y_4_comparing']
         )
    print(100*'_')

<function GradientBoostingCVModel at 0x7f4f23fe9730>
GradientBoostingRegressor(alpha=0.9, criterion='friedman_mse', init=None,
             learning_rate=0.005, loss='ls', max_depth=None,
             max_features=None, max_leaf_nodes=None,
             min_impurity_split=1e-07, min_samples_leaf=1,
             min_samples_split=2, min_weight_fraction_leaf=0.0,
             n_estimators=6000, presort='auto', random_state=None,
             subsample=1.0, verbose=0, warm_start=False) 


 421.20889837 342.771761754 1532.34615385
____________________________________________________________________________________________________


#>>>>>>>>>>>


# Final Model : Boosting


<function GradientBoostingCVModel at 0x7f4f23fe9730>
GradientBoostingRegressor(alpha=0.9, criterion='friedman_mse', init=None,
             learning_rate=0.005, loss='ls', max_depth=None,
             max_features=None, max_leaf_nodes=None,
             min_impurity_split=1e-07, min_samples_leaf=1,
             min_samples_split=2, min_weight_fraction_leaf=0.0,
             n_estimators=6000, presort='auto', random_state=None,
             subsample=1.0, verbose=0, warm_start=False) 


 421.20889837 342.771761754 1532.34615385

<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:><:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>
<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:><:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>
<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:><:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>
<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:><:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>


# 4 Random Forest

In [90]:
t_rforest = wrapper([RandomForestCVModel])


Dataset size read: train 256 and test 65 

# Tuning hyper-parameters 

Starting grdsurch.fit(X_train, y_train)

Best parameters set found on development set:

{'n_estimators': 4000}
RandomForestRegressor(bootstrap=True, criterion='mse', max_depth=None,
           max_features='auto', max_leaf_nodes=None,
           min_impurity_split=1e-07, min_samples_leaf=1,
           min_samples_split=2, min_weight_fraction_leaf=0.0,
           n_estimators=4000, n_jobs=-1, oob_score=True, random_state=None,
           verbose=0, warm_start=False)

【【】】adding new model for datasetIPPD.pickle 

【【【】】】newScore: 413.344478 
____________________________________________________________________________________________________


In [91]:
for key in t_rforest.keys():
    print(key)
    print(
          t_rforest[key]['model'],'\n\n\n',
          t_rforest[key]['train_rmse_cv_4_picking'],
          t_rforest[key]['test_rmse_4_reporting'],
          t_rforest[key]['test_mean_y_4_comparing']
         )
    print(100*'_')

<function RandomForestCVModel at 0x7f4f23fe97b8>
RandomForestRegressor(bootstrap=True, criterion='mse', max_depth=None,
           max_features='auto', max_leaf_nodes=None,
           min_impurity_split=1e-07, min_samples_leaf=1,
           min_samples_split=2, min_weight_fraction_leaf=0.0,
           n_estimators=4000, n_jobs=-1, oob_score=True, random_state=None,
           verbose=0, warm_start=False) 


 413.344478118 339.417223286 1532.34615385
____________________________________________________________________________________________________


#>>>>>>>>>>>


# Final Model: Random Forest

RandomForestRegressor(bootstrap=True, criterion='mse', max_depth=None,
           max_features='auto', max_leaf_nodes=None,
           min_impurity_split=1e-07, min_samples_leaf=1,
           min_samples_split=2, min_weight_fraction_leaf=0.0,
           n_estimators=4000, n_jobs=-1, oob_score=True, random_state=None,
           verbose=0, warm_start=False) 


 413.344478118 339.417223286 1532.34615385

<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:><:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>
<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:><:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>
<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:><:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>
<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:><:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>


RidgeCVModel, ElasticNetCVModel, 
                 SVRPolyCVModel, SVRSigmoidCVModel, SVRLinearCVModel, 
                 SVRRbfCVModel ] #OMPCVModel

# RidgeModel

In [113]:
t_ridge = wrapper([RidgeCVModel])


Dataset size read: train 256 and test 65 

# Tuning hyper-parameters 

Starting grdsurch.fit(X_train, y_train)

Best parameters set found on development set:

{'alpha': 1e-10}
Ridge(alpha=1e-10, copy_X=True, fit_intercept=True, max_iter=None,
   normalize=False, random_state=None, solver='auto', tol=1e-20)

【【】】adding new model for datasetIPPD.pickle 

【【【】】】newScore: 570.369987 
____________________________________________________________________________________________________


In [114]:
for key in t_ridge.keys():
    print(key)
    print(
          t_ridge[key]['model'],'\n\n\n',
          t_ridge[key]['train_rmse_cv_4_picking'],
          t_ridge[key]['test_rmse_4_reporting'],
          t_ridge[key]['test_mean_y_4_comparing']
         )
    print(100*'_')

<function RidgeCVModel at 0x7f4f23fe9840>
Ridge(alpha=1e-10, copy_X=True, fit_intercept=True, max_iter=None,
   normalize=False, random_state=None, solver='auto', tol=1e-20) 


 570.369986755 478.511724276 1532.34615385
____________________________________________________________________________________________________


#>>>>>>>>>>>


# Ridge - Final Model

Ridge(alpha=1e-10, copy_X=True, fit_intercept=True, max_iter=None,
   normalize=False, random_state=None, solver='auto', tol=1e-20) 


 570.369986755 478.511724276 1532.34615385

<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:><:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>
<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:><:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>
<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:><:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>
<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:><:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>


# Elastic Net

In [161]:
t_elasticNet = wrapper([ElasticNetCVModel])


Dataset size read: train 256 and test 65 

# Tuning hyper-parameters 

Starting grdsurch.fit(X_train, y_train)


/data/home/a1singh/anaconda3/lib/python3.6/site-packages/sklearn/linear_model/coordinate_descent.py:484: ConvergenceWarning: Objective did not converge. You might want to increase the number of iterations. Fitting data with very small alpha may cause precision problems.
  ConvergenceWarning)
/data/home/a1singh/anaconda3/lib/python3.6/site-packages/sklearn/linear_model/coordinate_descent.py:484: ConvergenceWarning: Objective did not converge. You might want to increase the number of iterations. Fitting data with very small alpha may cause precision problems.
  ConvergenceWarning)
/data/home/a1singh/anaconda3/lib/python3.6/site-packages/sklearn/linear_model/coordinate_descent.py:484: ConvergenceWarning: Objective did not converge. You might want to increase the number of iterations. Fitting data with very small alpha may cause precision problems.
  ConvergenceWarning)
/data/home/a1singh/anaconda3/lib/python3.6/site-packages/sklearn/linear_model/coordinate_descent.py:484: ConvergenceWarni


Best parameters set found on development set:

{'alpha': 1e-10, 'l1_ratio': 0.1}
ElasticNet(alpha=1e-10, copy_X=True, fit_intercept=True, l1_ratio=0.1,
      max_iter=10000000.0, normalize=False, positive=False,
      precompute=False, random_state=None, selection='cyclic', tol=1e-20,
      warm_start=False)

【【】】adding new model for datasetIPPD.pickle 

【【【】】】newScore: 582.076447 
____________________________________________________________________________________________________


/data/home/a1singh/anaconda3/lib/python3.6/site-packages/sklearn/linear_model/coordinate_descent.py:484: ConvergenceWarning: Objective did not converge. You might want to increase the number of iterations. Fitting data with very small alpha may cause precision problems.
  ConvergenceWarning)


In [ ]:
for key in t_elasticNet.keys():
    print(key)
    elasticNet_mo = t_elasticNet[key]['model']
    print(
          t_elasticNet[key]['model'],'\n\n\n',
          t_elasticNet[key]['train_rmse_cv_4_picking'],
          t_elasticNet[key]['test_rmse_4_reporting'],
          t_elasticNet[key]['test_mean_y_4_comparing']
         )
    print(100*'_')

In [163]:
#>>>>>>>>>>>

In [172]:
t_elasticNet = wrapper([ElasticNetCVModel])


Dataset size read: train 256 and test 65 

# Tuning hyper-parameters 

Starting grdsurch.fit(X_train, y_train)


/data/home/a1singh/anaconda3/lib/python3.6/site-packages/sklearn/linear_model/coordinate_descent.py:484: ConvergenceWarning: Objective did not converge. You might want to increase the number of iterations. Fitting data with very small alpha may cause precision problems.
  ConvergenceWarning)
/data/home/a1singh/anaconda3/lib/python3.6/site-packages/sklearn/linear_model/coordinate_descent.py:484: ConvergenceWarning: Objective did not converge. You might want to increase the number of iterations. Fitting data with very small alpha may cause precision problems.
  ConvergenceWarning)
/data/home/a1singh/anaconda3/lib/python3.6/site-packages/sklearn/linear_model/coordinate_descent.py:484: ConvergenceWarning: Objective did not converge. You might want to increase the number of iterations. Fitting data with very small alpha may cause precision problems.
  ConvergenceWarning)
/data/home/a1singh/anaconda3/lib/python3.6/site-packages/sklearn/linear_model/coordinate_descent.py:484: ConvergenceWarni


Best parameters set found on development set:

{'alpha': 1e-10, 'l1_ratio': 0.01}
ElasticNet(alpha=1e-10, copy_X=True, fit_intercept=True, l1_ratio=0.01,
      max_iter=10000000.0, normalize=False, positive=False,
      precompute=False, random_state=None, selection='cyclic', tol=1e-20,
      warm_start=False)

【【】】adding new model for datasetIPPD.pickle 

【【【】】】newScore: 583.860250 
____________________________________________________________________________________________________


/data/home/a1singh/anaconda3/lib/python3.6/site-packages/sklearn/linear_model/coordinate_descent.py:484: ConvergenceWarning: Objective did not converge. You might want to increase the number of iterations. Fitting data with very small alpha may cause precision problems.
  ConvergenceWarning)


In [173]:
for key in t_elasticNet.keys():
    print(key)
    elasticNet_mo = t_elasticNet[key]['model']
    print(
          t_elasticNet[key]['model'],'\n\n\n',
          t_elasticNet[key]['train_rmse_cv_4_picking'],
          t_elasticNet[key]['test_rmse_4_reporting'],
          t_elasticNet[key]['test_mean_y_4_comparing']
         )
    print(100*'_')

<function ElasticNetCVModel at 0x7f4f23fe98c8>
ElasticNet(alpha=1e-10, copy_X=True, fit_intercept=True, l1_ratio=0.01,
      max_iter=10000000.0, normalize=False, positive=False,
      precompute=False, random_state=None, selection='cyclic', tol=1e-20,
      warm_start=False) 


 583.860250045 498.000647708 1532.34615385
____________________________________________________________________________________________________


<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:><:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>
<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:><:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>
<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:><:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>
<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:><:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>


# SVR - Poly Kernel

In [ ]:
t_SVRPolyCVModel = wrapper([SVRPolyCVModel])

In [ ]:
t = t_SVRPolyCVModel

for key in t.keys():
    print(key)
    t_mo= t[key]['model']
    print(
          t[key]['model'],'\n\n\n',
          t[key]['train_rmse_cv_4_picking'],
          t[key]['test_rmse_4_reporting'],
          t[key]['test_mean_y_4_comparing']
         )
    print(100*'_')

# SVR - Poly Kernel - Final Model

#>>>>>>>>>>>

# SVR - Sigmoid Kernel

In [ ]:
t_SVRSigmoidCVModel = wrapper([SVRSigmoidCVModel])

In [ ]:
t = SVRSigmoidCVModel

for key in t.keys():
    print(key)
    t_mo= t[key]['model']
    print(
          t[key]['model'],'\n\n\n',
          t[key]['train_rmse_cv_4_picking'],
          t[key]['test_rmse_4_reporting'],
          t[key]['test_mean_y_4_comparing']
         )
    print(100*'_')

# SVR - Sigmoid Kernel - Final Model

#>>>>>>>>>>>


# SVR - Linear Kernel

In [ ]:
t_SVRLinearCVModel = wrapper([SVRLinearCVModel])

In [ ]:
t = SVRLinearCVModel

for key in t.keys():
    print(key)
    t_mo= t[key]['model']
    print(
          t[key]['model'],'\n\n\n',
          t[key]['train_rmse_cv_4_picking'],
          t[key]['test_rmse_4_reporting'],
          t[key]['test_mean_y_4_comparing']
         )
    print(100*'_')

# SVR - Linear Kernel - Final Model

#>>>>>>>>>>>


# SVR - rbf Kernel

In [ ]:
t_SVRRbfCVModel = wrapper([SVRRbfCVModel])

In [ ]:
t = t_SVRRbfCVModel

for key in t.keys():
    print(key)
    t_mo= t[key]['model']
    print(
          t[key]['model'],'\n\n\n',
          t[key]['train_rmse_cv_4_picking'],
          t[key]['test_rmse_4_reporting'],
          t[key]['test_mean_y_4_comparing']
         )
    print(100*'_')

# SVR - rbf Kernel - Final Model

#>>>>>>>>>>>


# (4C4) Response when four of four modules are chosen as training dataset

'''
### LassoCVModel at 0x7f64208e7c80
1025.08609776 880.826424842 1532.34615385
____________________________________________________________________________________________________
### GradientBoostingCVModel at 0x7f63fa3b07b8
421.108424701 342.356068455 1532.34615385
____________________________________________________________________________________________________
### RandomForestCVModel at 0x7f63fa3b0840
420.621205322 342.402960902 1532.34615385
____________________________________________________________________________________________________
### RidgeCVModel at 0x7f63fa3b08c8
1025.08583073 880.826144692 1532.34615385
____________________________________________________________________________________________________
### ElasticNetCVModel at 0x7f63fa3b0950
1025.08609776 880.826424842 1532.34615385
____________________________________________________________________________________________________
### SVRPolyCVModel at 0x7f63fa3b09d8
1032.95723148 882.068031346 1532.34615385
____________________________________________________________________________________________________
### SVRSigmoidCVModel at 0x7f63fa3b0a60
1080.46665239 1059.84229153 1532.34615385
____________________________________________________________________________________________________
### SVRLinearCVModel at 0x7f63fa3b0ae8
1080.46665239 1059.84229153 1532.34615385
____________________________________________________________________________________________________
### SVRRbfCVModel at 0x7f63fa3b0b70
1080.46665239 1059.84229153 1532.34615385
'''

In [ ]:
import pickle

with open('model_performance_all_modules.pickle', 'wb') as handle:
    pickle.dump(t, handle, protocol=pickle.HIGHEST_PROTOCOL)


In [ ]:
for mem in t.keys():
    print(100*'_')
    print(t[mem])
    print(100*'_')

In [ ]:
t = wrapper()
for mem in t.keys():
    print(100*'_')
    print(t[mem])
    print(100*'_')